src/dash_app.py

```diff
@@ -0,0 +1,378 @@
+                html.H2(
+                    children='Overlay of Segments with Original',
+                    style={'textAlign': 'left'}
+                ),
+                html.Div(
+                    children=[
+                        html.Img(
+                            src=data['ximage_b64'],
+                            style={
+                                'height': '50%',
+                                'width': '50%'
+                            }
+                        ),
+                        html.Img(
+                            src=data['yimage_b64'],
+                            style={
+                                'position': 'absolute',
+                                'top': 0,
+                                'left': 0,
+                                'opacity': 0.5,
+                                'height': '98%',
+                                'width': '50%'
+                            }
+                        )
+                    ],
+                    style={'position': 'relative'}
+                )
+            ],
+            style={'textAlign': 'left'}
+        ),
+    ])
+
+
+@dash_app.callback(
+    Output('labeled_pred', 'src'),
+    [Input('size_distr_json', 'children')]
+)
+def show_labeled_pred(size_distr_json):
+    """Displays labeled prediction image"""
+    data = json.loads(size_distr_json)
+    return data['rgb_pred_b64']
+
+
+@dash_app.callback(
+    Output('size_distr_graph', 'figure'),
+    [Input('size_distr_json', 'children')]
+)
+def update_hist(size_distr_json):
+    """Displays histogram of size distribution"""
+    data = json.loads(size_distr_json)
+    size_distr = np.asarray(data['size_distr_list'])
+    return {
+        'data': [go.Histogram(
+                    x=size_distr,
+                    xbins={'size': 1}
+                )],
+        'layout': go.Layout(
+            title={
+                'text': '<b>Size<br>Distribution</b>',
+                'font': {
+                    'size': 28
+                }
+            },
+            xaxis={'title': 'Size (pixels)'},
+            yaxis={'title': 'Count'},
+            annotations=[
+                go.layout.Annotation(
+                    text=(
+                        f'<b>Mean</b> = {size_distr.mean():.2E}<br><br>'
+                        f'<b>Median</b> = {np.median(size_distr):.2E}<br><br>'
+                        f'<b>Std</b> = {size_distr.std():.2E}'
+                    ),
+                    font={
+                        'size': 12
+                    },
+                    align='left',
+                    bordercolor='#B02405',
+                    borderwidth=2,
+                    borderpad=8,
+                    showarrow=False,
+                    xref='paper',
+                    yref='paper',
+                    x=0.02,
+                    y=1.30
+                )
+            ]
+        )
+    }
+
+
+if __name__ == '__main__':
+    dash_app.run_server(debug=True)
```

Comment:
This file is ridiculously long. Consider refactoring the functions into new files, and keeping *only* the UI functionality here.